### PR TITLE
Created CDK datepicker events and datepicker disabled examples

### PR DESCRIPTION
**Pull request by victoriaaa234**

### CODE REVIEW

File: src/material-examples/cdk-datepicker-disabled/cdk-datepicker-disabled-example.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+})
+export class CdkDatepickerDisabledExample {}
+
+
+@Component({
+  selector: 'my-disabled-calendar',
+  template: '',
+  providers: [{provide: CalendarView, useExisting: MyDisabledCalendar}],
+})
+export class MyDisabledCalendar<Date> extends CalendarView<Date> {
```
Comment (mmalerba) [inline src/material-examples/cdk-datepicker-disabled/cdk-datepicker-disabled-example.ts]: just use a `MatCalendar` or copy the `MyCalendar` from your other class, this class is a bad example

File: src/material-examples/cdk-datepicker-events/cdk-datepicker-events-example.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  }
+}
+
+
+@Component({
+  selector: 'my-events-calendar',
+  template: '',
+  providers: [{provide: CalendarView, useExisting: MyEventsCalendar}],
+})
+export class MyEventsCalendar<Date> extends CalendarView<Date> {
```
Comment (mmalerba) [inline src/material-examples/cdk-datepicker-events/cdk-datepicker-events-example.ts]: again, this is not a good example

File: src/material-examples/cdk-datepicker-disabled/cdk-datepicker-disabled-example.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import {Component} from '@angular/core';
+import {CalendarView} from '@angular/cdk/datetime';
+
+/** @title CDK Disabled datepicker */
```
Comment (jelbourn) [inline src/material-examples/cdk-datepicker-disabled/cdk-datepicker-disabled-example.ts]: Is `disabled` a complex enough use-case that it warrants its own example? I would think that you'd just need one main example of custom calendar with `CdkDatepicker` that has options for disabling and could show a snackbar or similar when there's an event.

Comment (victoriaaa234) [reply src/material-examples/cdk-datepicker-disabled/cdk-datepicker-disabled-example.ts]: The custom calendar doesn't have access to `CdkDatepicker` and whether or not the datepicker gets disabled

File: src/material-examples/cdk-datepicker-overview/cdk-datepicker-overview-example.ts [hunk 4]
```diff
@@ -27,29 +38,27 @@ export class CdkDatepickerOverviewExample<D> {
...
 
-  constructor(private _dateAdapter: DateAdapter<D>) {
-    super();
-    this.activeDate = this._dateAdapter.today();
-  }
-
   _selected(date: D) {
-    this.selected = date;
-    this.selectedChange.emit(date);
+    if (this.disabled) {} else {
```
Comment (jelbourn) [inline src/material-examples/cdk-datepicker-overview/cdk-datepicker-overview-example.ts]: `if (!this.disabled) {`

**Pull request merged**